File: JavaProject/GrowNet/src/main/java/ai/nektron/grownet/RegionBus.java
package ai.nektron.grownet;

public final class RegionBus {
    private double inhibitionFactor = 1.0;  // 1.0 = no inhibition
    private double modulationFactor = 1.0;  // scales learning rate
    private long currentStep = 0L;

    public double getInhibitionFactor() { return inhibitionFactor; }
    public double getModulationFactor() { return modulationFactor; }
    public long getCurrentStep()       { return currentStep; }

    public void setInhibitionFactor(double value) { inhibitionFactor = value; }
    public void setModulationFactor(double value) { modulationFactor = value; }

    /** Advance one tick and reset one‑tick pulses. */
    public void decay() {
        inhibitionFactor = 1.0;
        modulationFactor = 1.0;
        currentStep += 1;
    }
}

File: src/python/region_bus.py
class RegionBus:
    def __init__(self):
        self._inhibition_factor = 1.0
        self._modulation_factor = 1.0

    def get_inhibition_factor(self):
        return self._inhibition_factor

    def set_inhibition_factor(self, v):
        self._inhibition_factor = float(v)

    def get_modulation_factor(self):
        return self._modulation_factor

    def set_modulation_factor(self, v):
        self._modulation_factor = float(v)

    def reset(self):
        self._inhibition_factor = 1.0
        self._modulation_factor = 1.0
